src/emoji.ts

```diff
@@ -100,15 +115,14 @@ export const EMOJI: Emoji[] = EMOJIBASE.map((emojiData) => {
   // 'emoji'. We therefore strip any variation chars from strings
   // both when building the map and when looking up.
   UNICODE_TO_EMOJI.set(stripVariation(emoji.unicode), emoji);
-
   if (emoji.emoticon) {
     // Add mapping from emoticon to Emoji object
-    // eslint-disable-next-line @typescript-eslint/no-unused-expressions
-    Array.isArray(emoji.emoticon)
-      ? emoji.emoticon.forEach((x) => EMOTICON_TO_EMOJI.set(x, emoji))
-      : EMOTICON_TO_EMOJI.set(emoji.emoticon, emoji);
+    const emoticons = [emoji.emoticon]
+      //flatten, in case `emoji.emoticon` is an array
+      .flat()
+      .flatMap((x) => generateEmoticonPermutations(x));
+    emoticons.forEach((x) => EMOTICON_TO_EMOJI.set(x, emoji));
```

Comment:
It's feel wrong to me to mutate a map/array when doing a map another array.

Reply:
Yes, true I found that strange also, I wasn't too comfortable making those logic changes before as there wasn't much tests but we have that now :). Will move the logic that isn't related to the map out to a foreach so it's a bit clearer.